### PR TITLE
Update for Symfony 2.8 deprecations

### PR DESCRIPTION
**Pull request by WouterJ**
Fixes https://github.com/sonata-project/SonataAdminBundle/issues/3248

### CODE REVIEW

**Comment by Soullivaneuh** [comment]
Thank you @WouterJ, could you please check failing tests?

**Comment by WouterJ** [comment]
@Soullivaneuh should be fixed now

**Comment by rande** [comment]
@WouterJ Can you squash your commits ?

File: Form/FormMapper.php [hunk 1]
```diff
@@ -80,7 +80,8 @@ public function add($name, $type = null, array $options = array(), array $fieldD
 
         // change `collection` to `sonata_type_native_collection` form type to
         // avoid BC break problems
-        if ($type == 'collection') {
+        if ($type == 'collection' || $type == 'Symfony\Component\Form\Extension\Core\Type\CollectionType') {
```
Comment (Soullivaneuh) [inline Form/FormMapper.php]: Please use strict comparison.

**Comment by WouterJ** [comment]
Fixed issues and squashed commits

**Comment by OskarStark** [comment]
:+1:

**Comment by IvixLabs** [comment]
After this commit i got follow error on my project:

Route pattern "/xxx/ecommerce/customer/{id}/customerorder/{id}/customerorderelement/list" cannot reference variable name "id" more than once.

Not sure where is problem.

I use ecommerce bundle.

**Comment by OskarStark** [comment]
@IvixLabs can you please open a new issue in the ecommerce repo?

thank you

**Pull request merged**